include/gatekeeper_mailbox.h

```diff
@@ -19,14 +19,19 @@
 #ifndef _GATEKEEPER_MAILBOX_H_
 #define _GATEKEEPER_MAILBOX_H_
 
-/*
- * TODO Remove this preprocessing directive when there are
- * are calls to these functions; for now, the mailbox is
- * just an example of how to set up a library.
- */
-#if 0
-void create_mailbox(void);
-void destroy_mailbox(void);
-#endif
+#include <rte_ring.h>
+#include <rte_mempool.h>
+
+struct mailbox {
+	struct rte_ring    *ring;
+	struct rte_mempool *pool;
+};
+
+struct mailbox *create_mailbox(
```

Comment:
What about renaming this function to `init_mailbox()`, and instead of allocating `struct mailbox`, `init_mailbox()` receives a pointer to that struct to initialize it? The idea is to reduce one indirection because we don't seem to need the indirection in the code, and each indirection is one more memory lookup. The caller can easily pack `struct mailbox` into whatever it's already allocating.

Before accepting this suggestion, check that it makes sense.